java/app/src/main/java/com/fullstorydev/shoppedemo/data/CustomerInfo.java

```diff
@@ -0,0 +1,52 @@
+package com.fullstorydev.shoppedemo.data;
+
+import com.fullstorydev.shoppedemo.utilities.Constants;
+
+import java.util.HashMap;
+
+public class CustomerInfo {
+    private HashMap<String,String> strMap;
+    private HashMap<String,Integer> intMap;
+
+    CustomerInfo(){
+        // all key-value pairs available for USER_INFO_STRING_KEYS
+        strMap = new HashMap<>();
+        // all key-value pairs available for USER_INFO_INTEGER_KEYS
+        intMap = new HashMap<>();
+    }
+
+    // all available user info keys are managed by Constants.USER_INFO_STRING_KEYS
+    boolean putValue(String key, String val){
+        for(String k: Constants.USER_INFO_STRING_KEYS){
+            if(k.equals(key)) {
+                strMap.put(key, val);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // all available user info keys are managed by Constants.USER_INFO_INTEGER_KEYS
+    boolean putValue(String key, Integer val){
+        for(String k: Constants.USER_INFO_INTEGER_KEYS){
+            if(k.equals(key)) {
+                intMap.put(key, val);
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
why use hash maps like this rather than use getters and setters?

Reply:
great question and exactly why I've been working on this for this long. 
The problem with having specific setters is that all of those methods need to propagate through model->repository->view model and finally used by the view via binding. It gets harder when using encapsulate the whole form into one LiveData, you'd have to make it "property aware". It would potentially make the code more readable but also likely make it a harder to maintain when fields need to be added/modified. I wanted to use the constants as a "catalog" of the fields and have them organized. 
Having said that I realize form validation might be a challenge this way. I have ruled out of two-way binding because the unnecessary calls to update model on each key stroke. But maybe a two-way binding is the solution after all... WDYT
FYI: https://developer.android.com/topic/libraries/data-binding/two-way

Reply:
I think erring on the side of readability is the right choice. I'm not sure what the consequences are of persisting data as it is entered. I skimmed through those articles and it seems like two-way binding on ViewModels is once approach that Android recommends, though having to verify that values have actually changed at the risk of incurring an infinite loop is a little freaky. I always prefer unidirectional binding for these types of reasons.

I'll go ahead and approve.